### DISCUSSION TITLE
web3.js的库怎么获取不到钱包余额了

### DISCUSSION THREAD

**Discussion by zheng2330** [consenlabs/webview#3] [General]
//查询用于usdt
    async queryusdt() {
      try {
        let tokenBalance = await usdtContract.methods
          .balanceOf(this.address)
          .call();
        let value = this.$fromWei(tokenBalance, "usdt");
       console.log(value)
      } catch (error) {
       console.log(error)
      }
    },

    //用于查询eth
    async queryeth() {
      try {
        let ethBalance = await Web3Provider.eth.getBalance(this.address);

        let value = this.$fromWei(ethBalance, "ether");
          console.log(value)
      } catch (error) {
       console.log(error)
      }
    },
    //用于查询dibi余额
    async querydibi() {
      try {
        let dibiBalance = await dibiContract.methods
          .balanceOf(this.address)
          .call();
        let value = this.$fromWei(dibiBalance, "Gwei");
          console.log(value)
      } catch (error) {
       console.log(error)
      }
    },
    //用于查询dbl余额
    async querydbl() {
      try {
        let dibiBalance = await dblContract.methods
          .balanceOf(this.address)
          .call();
        let value = this.$fromWei(dibiBalance, "Gwei");
        console.log(value)
      } catch (error) {
       console.log(error)

      }
    },

**Comment by unix** [comment]
这些都是从链上取的数据，和 imToken 没有联系。
imToken 提供标准的 EIP 接口，见 EIP 实现：https://imtoken.gitbook.io/developers/products/webview/rpcs

当前问题的上下文很少，无法判断问题原因。如果是未申请权限，请通过 EIP-1102 申请，如果是已获得地址但未查询到余额，需要检查应用本身的逻辑。
